src/backend/allocator/mod.rs

```diff
@@ -67,12 +69,13 @@ pub trait Allocator<B: Buffer> {
         height: u32,
         fourcc: Fourcc,
         modifiers: &[Modifier],
-    ) -> Result<B, Self::Error>;
+    ) -> Result<Self::Buffer, Self::Error>;
 }
 
 // General implementations for interior mutability.
 
-impl<A: Allocator<B>, B: Buffer> Allocator<B> for Arc<Mutex<A>> {
+impl<A: Allocator> Allocator for Arc<Mutex<A>> {
+    type Buffer = A::Buffer;
```

Comment:
Actually this brings up a good question. If a buffer implements AsDmabuf, exporting it might not be synchronized. In the vulkan allocator, we export directly in the buffer, but it is safe as we have an Arc<Device> inside.

However for other backends this may vary.

Reply:
Yeah right, but we don't do a particular good job at accounting for synchronization (which is also hard, when you have multiple objects - textures, dmabufs, framebuffer - referr to the same GPU memory). So I don't think this is an issue for this PR, but more something we should have a longer discussion about.